lib/wallet-e2e/exe/Main.hs

```diff
@@ -42,19 +47,49 @@ import Test.Syd
 
 main :: IO ()
 main = withUtf8 do
-    testNetworkOptions <-
+    TestOptions{..} :: TestOptions <-
         execParser
             $ info
-                (parser <**> helper)
+                (parserTestOptions <**> helper)
                 (fullDesc <> progDesc "E2E Wallet test suite")
     testNetwork <- testNetworkOptionsToConfig testNetworkOptions
+    traceConfiguration <- do
+        absTraceDir <-
+            traverse makeDirAbsolute
+                $ testGlobalOptionsTraceOutput testGlobalOptions
+        pure $ TraceConfiguration absTraceDir
     sydTestWith SydTest.defaultSettings{SydTest.settingRetries = 1} do
         effectsSpec
-        walletSpec testNetwork
+        walletSpec traceConfiguration testNetwork
+
+parserTestOptions :: Parser TestOptions
+parserTestOptions = TestOptions <$> parserNetworkOptions <*> parserGlobalOptions
```

Comment:
Lets move this parser right below the `TestOptions` data type inside the -- Command line options --- section